src/meltano/core/state_service.py

```diff
@@ -16,6 +16,8 @@
 from meltano.core.job import Job, JobFinder, Payload, State
 from meltano.core.utils import merge
 
+STATE_ID_COMPONENT_DELIMITER = ":"
```

Comment:
Guessing pulling STATE_ID_COMPONENT_DELIMETER and `meltano.core.block.extract_load.generate_state_id` into meltano.core.utils or the like probably doesn't help ?

Reply:
I'll have a look to see if that works. Would that be preferable?

Reply:
Thinking about , I bet you'd still end up with a circular import. The annoy bit is that generate_state_id wants a consumer/producer IOBlock , but *only* uses the string_id's of those. One way to clean this up then would be to update `generate_state_id` to just expect a string for consumer/producer and just have callers start passing the string_id directly rather than passing the whole IOBlock.

That's a bit more work/refactoring though - I think you could keep the pr as-is and then we can log a follow up issue to clean up/refactor.

Reply:
Sure :+1: